samples/matter/common/src/ota_util.cpp

```diff
@@ -21,27 +21,33 @@ DefaultOTARequestorDriver sOTARequestorDriver;
 chip::BDXDownloader sBDXDownloader;
 chip::DefaultOTARequestor sOTARequestor;
 
-} /* namespace */
-
-/* compile-time factory method */
-OTAImageProcessorImpl &GetOTAImageProcessor()
-{
+/**
+ * Select recommended OTA image processor implementation.
+ *
+ * If the application uses QSPI external flash and enables API for controlling
+ * power states of peripherals, select the implementation that automatically
+ * powers off the external flash when no longer needed. Otherwise, select the
+ * most basic implementation.
+ */
 #if CONFIG_PM_DEVICE && CONFIG_NORDIC_QSPI_NOR
-	static ExtFlashHandler sQSPIHandler;
-	static OTAImageProcessorImplPMDevice sOTAImageProcessor{ sQSPIHandler };
+ExtFlashHandler sQSPIHandler;
+OTAImageProcessorImplPMDevice sOTAImageProcessor{ sQSPIHandler };
 #else
-	static OTAImageProcessorImpl sOTAImageProcessor;
+OTAImageProcessorImpl sOTAImageProcessor;
 #endif
-	return sOTAImageProcessor;
-}
+
+} /* namespace */
 
 void InitBasicOTARequestor()
 {
-	OTAImageProcessorImpl &imageProcessor = GetOTAImageProcessor();
-	imageProcessor.SetOTADownloader(&sBDXDownloader);
-	sBDXDownloader.SetImageProcessorDelegate(&imageProcessor);
+	sOTAImageProcessor.SetOTADownloader(&sBDXDownloader);
+	sBDXDownloader.SetImageProcessorDelegate(&sOTAImageProcessor);
 	sOTARequestorStorage.Init(Server::GetInstance().GetPersistentStorage());
 	sOTARequestor.Init(Server::GetInstance(), sOTARequestorStorage, sOTARequestorDriver, sBDXDownloader);
-	sOTARequestorDriver.Init(&sOTARequestor, &imageProcessor);
+	sOTARequestorDriver.Init(&sOTARequestor, &sOTAImageProcessor);
 	chip::SetRequestorInstance(&sOTARequestor);
+
+#if CONFIG_PM_DEVICE && CONFIG_NORDIC_QSPI_NOR
+	sOTAImageProcessor.PutPMDeviceSleep();
```

Comment:
Where is the `PutPMDeviceSleep` defined? I cannot find it anywhere in the Github diff of this commit :)

Reply:
It is in the corresponding commit to the sdk-connectedhomeip fork: https://github.com/nrfconnect/sdk-connectedhomeip/pull/65/files#diff-58afcbe9511e695bb69ec55703ca86add9bd5c890fac32ce61cdcb1f103f2782R223